### ISSUE TITLE
GU0034 should not warn here

### ISSUE THREAD

**Issue by JohanLarsson** [JohanLarsson/Gu.Analyzers#89]
```c#
        public static IReadOnlyList<EventArgs> ResetArgs()
        {
            var reference = new ObservableCollection<int> { 1 };
            using (var events = reference.SubscribeObservableCollectionEvents())
            {
                reference.Clear();
                return events;
            }
        }
```

returning a disposed object is weird but not the job of this analyzer.
